docs/argument_processing.rst

```diff
@@ -371,7 +371,7 @@ argument list instead of a string::
             # cmdline contains a string
             pass
 
-        @with_argument_list
+        @with_argument_list()
```

Comment:
I find the required parentheses without any argument slightly awkward, but not a big deal.  Just a thought - is it with having the preserve_quotes argument at all?  Are there important use cases we would miss by just always preserving quotes when this decorator is used?